modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultPolicyInstance.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.deployment.impl.internal.policy;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyList;
+import static org.mule.runtime.core.config.bootstrap.ArtifactType.APP;
+import static org.mule.runtime.module.deployment.impl.internal.artifact.ArtifactContextBuilder.newBuilder;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.core.api.config.MuleProperties;
+import org.mule.runtime.core.api.registry.RegistrationException;
+import org.mule.runtime.core.policy.Policy;
+import org.mule.runtime.core.policy.PolicyParametrization;
+import org.mule.runtime.core.policy.PolicyPointcut;
+import org.mule.runtime.core.policy.PolicyPointcutParameters;
+import org.mule.runtime.deployment.model.api.application.Application;
+import org.mule.runtime.deployment.model.api.artifact.ArtifactContext;
+import org.mule.runtime.deployment.model.api.policy.PolicyTemplate;
+import org.mule.runtime.module.artifact.classloader.ClassLoaderRepository;
+import org.mule.runtime.module.deployment.impl.internal.artifact.ArtifactContextBuilder;
+import org.mule.runtime.module.service.ServiceRepository;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Default implementation of {@link PolicyInstance} that depends on a {@link PolicyTemplate} artifact.
+ */
+public class DefaultPolicyInstance implements PolicyInstance {
+
+  private ArtifactContext policyContext;
+  private final Application application;
+  private final PolicyTemplate template;
+  private final PolicyParametrization parametrization;
+  private final ServiceRepository serviceRepository;
+  private final ClassLoaderRepository classLoaderRepository;
+  private org.mule.runtime.core.policy.PolicyInstance policyInstance;
```

Comment:
Yeah... I know is weird. 
This is class not a wrapper of org.mule.runtime.core.policy.PolicyInstance. There are duplicated concepts and at this point I don't know which one has to be renamed. I think that the PolicyInstance from core should not be there or should be renamed, but I'm not sure at this point.